### ISSUE TITLE
Add dtype_dict to problem definition

### ISSUE THREAD

**Issue by paxcema** [mindsdb/lightwood#906]
This enables easy type inference overriding.

**Comment by infinity1729** [comment]
Hey @paxcema, I want to give this a try. Can you please elaborate on what I am suppose to do, and what is the objective of dtype_dict

**Comment by paxcema** [comment]
Hi @infinity1729,

`dtype_dict` is a dictionary that contains the inferred data types for each column observed at training time. For example:

```python
dtype_dict = {
    'n_bathrooms': 'categorical',
    'price': 'integer',
    ...
}
```

To add this explicitly in the ProblemDefinition class, you need to make the necessary changes in `lightwood/api/types.py`.

We also need to figure out if there are any additional actions required to actually override the inferred dtypes with custom choices, so you should also write a test for this where the problem definition class is instanced with a particular `dtype_dict`, train some model, and check that the dispatched encoder actually corresponds to the custom `dtype`.

**Comment by paxcema** [comment]
Closed via #1064.
